lib/handshake_client13.ml

```diff
@@ -20,24 +20,27 @@ let answer_server_hello state ch (sh : server_hello) secrets raw log =
       | None -> Error (`Fatal `InvalidServerHello)
       | Some (_, secret) ->
         let* shared = Handshake_crypto13.dh_shared secret share in
-        let hlen = Mirage_crypto.Hash.digest_size (Ciphersuite.hash13 cipher) in
+        let hlen =
+          let module H = (val Digestif.module_of_hash' (Ciphersuite.hash13 cipher)) in
+          H.digest_size
+        in
         let* psk, resumed =
           match
             Utils.map_find ~f:(function `PreSharedKey idx -> Some idx | _ -> None) sh.extensions,
             state.config.Config.cached_ticket
           with
-          | None, _ | _, None -> Ok (Cstruct.create hlen, false)
+          | None, _ | _, None -> Ok (String.make hlen '\x00', false)
           | Some idx, Some (psk, _epoch) ->
             let* () = guard (idx = 0) (`Fatal `InvalidServerHello) in
             Ok (psk.secret, true)
         in
         let early_secret = Handshake_crypto13.(derive (empty cipher) psk) in
         let hs_secret = Handshake_crypto13.derive early_secret shared in
-        let log = log <+> raw in
+        let log = log ^ raw in
```

Comment:
if I remember, we should be able to use a hash state since we never need the entire log data. this is for sure a separate PR.